File: apps/api/internal/models/template.go
package models

type Template struct {
	BaseModel
	Name           string `gorm:"size:200;not null" json:"name"`
	Description    string `gorm:"type:text" json:"description,omitempty"`
	ProjectType    string `gorm:"size:50;not null" json:"project_type"`
	FrontendStack  string `gorm:"size:200" json:"frontend_stack,omitempty"`
	BackendStack   string `gorm:"size:200" json:"backend_stack,omitempty"`
	DatabaseStack  string `gorm:"size:200" json:"database_stack,omitempty"`
	AIStack        string `gorm:"size:200" json:"ai_stack,omitempty"`
	UIStyle        string `gorm:"size:200" json:"ui_style,omitempty"`
	DefaultPrompts string `gorm:"type:text" json:"default_prompts,omitempty"`
	DefaultRules   string `gorm:"type:text" json:"default_rules,omitempty"`
}

File: frontend/src/api/template.ts
import api from './client'
import type { ApiResponse } from './types'

export interface Template {
  id: string
  name: string
  description: string
  project_type: string
  frontend_stack: string
  backend_stack: string
  database_stack: string
  ai_stack: string
  ui_style: string
  default_prompts: string
  default_rules: string
  created_at: string
  updated_at: string
}

export interface TemplateListResponse {
  templates: Template[]
  total: number
}

export const templateApi = {
  list(params?: { project_type?: string; page?: number; page_size?: number }) {
    return api.get<ApiResponse<TemplateListResponse>>('/templates', { params })
  },
  get(id: string) {
    return api.get<ApiResponse<Template>>(`/templates/${id}`)
  },
}
